engine/src/cmd/collide2/Ice/IceFPU.h

```diff
@@ -14,17 +14,24 @@
 
 	#define	SIGN_BITMASK			0x80000000
 
+    union ice_union
+    {
+        ice_sdword sdword_member;
+        ice_udword udword_member;
+        float      float_member;
+    };
+
 	//! Integer representation of a floating-point value.
-	#define IR(x)					((udword&)(x))
+	#define IR(x)					((static_cast<ice_union*>((void *)&(x)))->udword_member)
```

Comment:
Another key set of changes. Before, everywhere these four macros were used (and they are used a LOT), they were triggering warnings about type-punning with pointers violating strict aliasing, or something like that. A previous developer suppressed these warnings with a `#pragma` GCC directive in Stdafx.h. Now the warnings are no longer suppressed. They don't need to be, because the code is more correct. :-)

Reply:
Q: do we need versions for pointers and versions for non-pointers? (Just curious)
Also, might it make sense to change these from C Macros to C++ inline methods?

Reply:
@BenjamenMeyer no, we don't need separate versions for pointers. They wouldn't be used anywhere.

It might make sense to change these to C++ inline methods at some point, but that seems like a potentially risky move. One that I'm not ready to make in this PR.

Reply:
Changing to C++ inline methods shouldn't make a difference for risk; but would expose places where they're changed as linting would be a bit more effective.